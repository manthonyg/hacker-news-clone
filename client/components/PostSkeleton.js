import React from 'react';
import styled, { keyframes } from 'styled-components';

const loadingAnimation = keyframes`
100% {
    transform: translateX(100%);
}
`;

const Skeleton = styled.div`
  background: white;
  border-radius: 0.25rem;
  width: 98vw;
  padding: 10px;
`;
const Loading = styled.div`
  position: relative;
  background-color: #e2e2e2;
  &::after {
    display: block;
    content: '';
    top: 0;
    position: absolute;
    width: 100%;
    height: 100%;
    transform: translateX(-100%);
    background: linear-gradient(
      90deg,
      transparent,
      rgba(255, 255, 255, 0.5),
      transparent
    );
    animation: ${loadingAnimation} 2s infinite;
  }
`;

const SkeletonInner = styled.div`
  height: 35px;
`;

function PostSkeleton({ numberOfSkeletons }) {
  return [...Array(numberOfSkeletons)].map((item, i) => {
    return (
      <Skeleton key={Date.now() + i}>
        <Loading>
          <SkeletonInner />
        </Loading>
      </Skeleton>
    );
  });
}

export default PostSkeleton;
