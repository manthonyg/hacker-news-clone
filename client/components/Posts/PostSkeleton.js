import React from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";

const PostCard = styled.div`
  display: flex;
  align-content: center;
  width: 100%;
  min-height: 100px;
  height: 100%;
  background-color: ${(props) => {
    if (props.theme.theme === "light") return "#FFFFFF";
    return "#282828";
  }};
  border-bottom: ${(props) => {
    if (props.theme.theme === "light") return "1px solid #c3c3c3";
    return "1px solid #404040";
  }};
`;

const withEmoji = css`
  content: "";
  position: absolute;
  border-radius: 100%;
  left: -50px;

  width: 40px;
  height: 40px;
  background-color: ${(props) => {
    if (props.theme.theme === "light") return "#eeeeee";
    return "#212121";
  }};
`;

const Skeleton = styled.div`
  display: flex;
  padding-left: 80px;
  width: 50vw;
`;

const SkeletonContent = styled.span`
  height: 13px;
  position: absolute;
  top: 40px;
  width: 100%;
  background-color: ${(props) => {
    if (props.theme.theme === "light") return "#eeeeee";
    return "#212121";
  }};

  &:nth-child(1):before {
    ${(props) => (props.noEmoji ? "" : withEmoji)}
  }
  &:nth-child(2) {
    height: 8px;
    top: 60px;
  }
`;

const Loading = styled.div`
  position: relative;
  width: 100%;
`;

const SkeletonInner = styled.div`
  height: 75px;
  margin-top: 20px;
`;

function PostSkeleton({ numberOfSkeletons = 1, noEmoji }) {
  return (
    <>
      {[...Array(numberOfSkeletons)].map((skeleton, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <PostCard key={i + numberOfSkeletons}>
          <Skeleton key={i}>
            <Loading>
              <SkeletonContent noEmoji={noEmoji} />
              <SkeletonContent />
              <SkeletonInner />
            </Loading>
          </Skeleton>
        </PostCard>
      ))}
    </>
  );
}

PostSkeleton.propTypes = {
  numberOfSkeletons: PropTypes.number.isRequired,
  noEmoji: PropTypes.bool,
};

export default PostSkeleton;
