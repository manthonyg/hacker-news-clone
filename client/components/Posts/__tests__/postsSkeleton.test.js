import React from "react";
import renderer from "react-test-renderer";
import PostSkeleton from "../PostSkeleton";

it("renders correctly with all props", () => {
  const requiredProps = {
    numberOfSkeletons: 1,
    noEmoji: true,
  };

  const tree = renderer
    .create(
      <>
        <PostSkeleton {...requiredProps} />
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
