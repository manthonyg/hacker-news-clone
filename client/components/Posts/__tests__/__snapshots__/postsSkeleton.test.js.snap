// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="sc-AxjAm iBeqQx"
>
  <div
    className="sc-AxirZ fnqyIr"
  >
    <div
      className="sc-AxhCb cgpiIf"
    >
      <span
        className="sc-AxiKw kVugzN"
      />
      <span
        className="sc-AxiKw cNuSwo"
      />
      <div
        className="sc-AxhUy fvWBGO"
      />
    </div>
  </div>
</div>
`;
